osquery/tables/system/darwin/keychain_acl.cpp

```diff
@@ -0,0 +1,334 @@
+                                                             app_index);
+        CFDataRef data = nullptr;
+        os_status = SecTrustedApplicationCopyData(app, &data);
+        if (os_status != 0) {
+          return Status(os_status, "Coult not copy trusted application data");
+        }
+
+        bytes = CFDataGetBytePtr(data);
+        if (bytes && bytes[0] == 0x2f) {
+          acl_data.applications.push_back(std::string((const char *)bytes));
+        }
+      }
+    }
+
+    acls.push_back(acl_data);
+  }
+
+  return Status(0, "OK");
+}
+
+static void bufferToStreamHex(std::stringstream &stream,
+                              UInt32 length,
+                              const void *data) {
+  uint8 *p = (uint8 *)data;
+  while (length--) {
+    int ch = *p++;
+    stream << boost::format("%02X") % ch;
+  }
+}
+
+static void bufferToStreamAscii(std::stringstream &stream,
+                                UInt32 length,
+                                const void *data) {
+  uint8 *p = (uint8 *)data;
+  while (length--) {
+    char ch = *p++;
+    if (ch >= ' ' && ch <= '~' && ch != '\\') {
+      stream << (char)ch;
+    } else {
+      stream << std::hex << ch;
+    }
+  }
+}
+
+void bufferToStream(std::stringstream &stream,
+                    UInt32 length,
+                    const void *data) {
+  uint8 *p = (uint8 *)data;
+  bool hex = false;
+  bool ascii = false;
+  for (int i = 0; i < length; ++i) {
+    int ch = *p++;
+    if (ch >= ' ' && ch <= '~' && ch != '\\') {
+      ascii = true;
+    } else {
+      hex = true;
+    }
+  }
+
+  if (hex) {
+    stream << "0x";
+    bufferToStreamHex(stream, length, data);
+    if (ascii) {
+      stream << " ";
+    }
+    stream << " ";
+  }
+  if (ascii) {
+    bufferToStreamAscii(stream, length, data);
+  }
+}
+
+std::string attributeBufferToString(const void *data, UInt32 length) {
+  std::stringstream stream;
+  bufferToStream(stream, length, data);
+  return stream.str();
+}
+
+Status genKeychainACLApps(const std::string &path, QueryData &results) {
+  SecKeychainRef keychain = nullptr;
+  OSStatus os_status = 0;
+
+  os_status = SecKeychainOpen(path.c_str(), &keychain);
+  if (os_status != 0) {
+    return Status(os_status, "Could not open the keychain at " + path);
+  }
+  if (keychain == nullptr) {
+    return Status(1, "keychain object was not populated properly");
+  }
+
+  SecKeychainSearchRef search = nullptr;
+  OSQUERY_USE_DEPRECATED(os_status = SecKeychainSearchCreateFromAttributes(
+                             keychain, CSSM_DL_DB_RECORD_ANY, NULL, &search););
+  if (os_status != 0) {
+    return Status(os_status,
+                  "Could not pull keychain items from the search API");
+  }
+  if (search == nullptr) {
+    return Status(1, "keychain search object was not populated properly");
+  }
+
+  SecKeychainItemRef item = nullptr;
+  OSQUERY_USE_DEPRECATED(while (SecKeychainSearchCopyNext(search, &item) == 0) {
```

Comment:
Wrapping this whole loop in the deprecated silence bugs me. I know having the iterator logic in the loop expression is ideal, but breaking the deprecated call into the next line alongside a break condition:

```cpp
OSQUERY_USE_DEPRECATED(os_status = SecKeychainSearchCopyNext(search, &item)); 
if (os_status != noErr) {
  break;
}
```